compiler/parser/parser.peg

```diff
@@ -466,6 +466,26 @@ SampleProc
                                 "rhs": MAP("kind": "ID", "name": "sample")))))))
     }
 
+// AmbiguousExprList is a top level operator that could either be 1) A
+// summarize proc 2) a put proc or 3) a filter. The semantic pass is responsible
+// for converting to either of the 3.
+AmbiguousExprList
+  = args:AmbiguousExprs {
+    RETURN(MAP("kind": "AmbiguousExprList", "args": args))
+  }
+
+AmbiguousExprs
+  = first:AmbiguousExpr rest:(__ "," __ expr:AmbiguousExpr { RETURN(expr) })* {
+    RETURN(PREPEND(first, rest))
+  }
+
+AmbiguousExpr
+ = Assignment
+// XXX Perhaps SearchFactor should be removed here and we leave it up to the 
+// semantic compiler to convert a raw primitive to a search?
+ / SearchFactor
+ / Expr
```

Comment:
I thought this would just be Assignment and Function... why do you need Expr and SearchFactor?

Reply:
I think I got ahead of myself and started thinking it would be nice if any expression without a proc should just get parsed as an AmbiguousExprList (or OpExrs) and relying on the semantic compiler to figure out the context (or throw an error if expr list doesn't make any sense), instead of having the parser figure out the context and return a parser error if the context is off. I still think this is the right approach, but not necessary in the scope of this change. I will change it as you suggest.